src/tracks/ui/CommonTrackPanelCell.cpp

```diff
@@ -44,6 +49,64 @@ HitTestPreview CommonTrackPanelCell::DefaultPreview
    return { {}, &defaultCursor, {} };
 }
 
+std::vector<ComponentInterfaceSymbol> CommonTrackPanelCell::GetMenuItems(
+   const wxRect&, const wxPoint *, AudacityProject * )
+{
+   return {};
+}
+
+unsigned CommonTrackPanelCell::DoContextMenu( const wxRect &rect,
+   wxWindow *pParent, const wxPoint *pPoint, AudacityProject *pProject)
+{
+   const auto symbols = GetMenuItems( rect, pPoint, pProject );
+   if (symbols.empty())
+      return RefreshCode::RefreshNone;
+
+   // Set up command context with extras
+   CommandContext context{ *pProject };
+   SelectedRegion region;
+   if (pPoint) {
+      auto time = ViewInfo::Get(*pProject).PositionToTime(pPoint->x, rect.x);
+      region = { time, time };
+      context.temporarySelection.pSelectedRegion = &region;
+   }
+   context.temporarySelection.pTrack = FindTrack().get();
+
+   auto &commandManager = CommandManager::Get(*pProject);
+   auto flags = MenuManager::Get( *pProject ).GetUpdateFlags();
+
+   // Common dispatcher for the menu items
+   auto dispatcher = [&]( wxCommandEvent &evt ){
+      auto idx = evt.GetId() - 1;
+      if (idx >= 0 && idx < symbols.size()) {
+         commandManager.HandleTextualCommand(
+            symbols[idx].Internal(), context, flags, false);
```

Comment:
As far as I can see, the call to dispatcher is deferred, if so it raises questions about captured objects lifetime

Reply:
You are right to raise the question.  And the answer is that the `[&]` capture is safe, because the popup menu, like a modal dialog, has a lifetime limited to the enclosing function.  When the menu is destroyed, the lambdas remembered by all the Bind() calls on in are also destroyed, and all before the function exits.

Reply:
Oh, I see now, PopupMenu "_returns control when the user has dismissed the menu_". Pointer to the SelectedRegion is safe too i presume?

Reply:
Why would it not be safe too?  All those variables also have lifetimes that include the lambda's lifetime.